flake.nix

```diff
@@ -64,14 +71,85 @@
               fourmolu
               haskell-language-server
               hlint
-            ])++ [ cardanoPkgs.cardano-node cardanoPkgs.cardano-cli ];
+            ])++ [ cardanoPkgs.cardano-node cardanoPkgs.cardano-cli ] ++ [pkgs.postgresql];
         });
-      testsNet = perSystem (system:
+      testnet = perSystem (system:
+        let pkgs =nixpkgsFor system;
+        in
+          pkgs.runCommand "setPrivateNetwork" {buildInputs=(with pkgs; [git]);} (builtins.readFile ./setPrivateNetwork.sh)
+        );
+      runTestnet = self.testnet.x86_64-linux.testnet;
+
+      testnet2 = perSystem (system: 
         let
           pkgs = nixpkgsFor system;
-        in self.packages.${system} // {
-          testnet = pkgs.runCommand "setPrivateNetwork" ''./setPrivateNetwork.sh'';
-        });
+          cardanoPkgs = cardanoPkgsFor system;
+          privateNetwork.url = "https://github.com/woofpool/cardano-private-testnet-setup";
+          localPath = "test-env/ogmios-datum-cache-private-network/cardano-private-testnet-setup";
+        in 
+          pkgs.writeShellScript "runTestnet" ''
```

Comment:
If you're going to write any more complex shell scripts directly in Nix, I'd recommend using `pkgs.writeShellApplication` instead. It automatically runs `shellcheck` on the provided script. Here's the manual entry for it: https://nixos.org/manual/nixpkgs/stable/#trivial-builder-writeShellApplication